docx2txt 1.4 (new formula)

proinsias opened the pull request:
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
    + <s>I'm getting an error about an "Empty installation"</s> (resolved)
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

<s>I've also added a test, but when I run it I get:</s>

<s>`Error: Testing requires the latest version of docx2txt` </s>
<s>`Check if </tmp/docx2txt-test-20160617-27321-1dgayqb/demo.docx> exists, is readable and has non-zero size!`</s>

<s>It seems as if the `demo.docx` is not copied to the `testpath`.

Any guidance on these issues would be appreciated.</s> (resolved)

dunn commented on Formula/docx2txt.rb:
```diff
@@ -0,0 +1,19 @@
+  end
+
+  test do
+    system "wget", "http://calibre-ebook.com/downloads/demos/demo.docx"
```
This should be a vendored `resource` to enable checksumming; see [`aften.rb`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/aften.rb) for an example.

dunn replied:
This should be a vendored `resource` to enable checksumming; see [`aften.rb`](https://github.com/Homebrew/homebrew-core/blob/master/Formula/aften.rb) for an example.

dunn commented on Formula/docx2txt.rb:
```diff
@@ -0,0 +1,19 @@
+
+  test do
+    system "wget", "http://calibre-ebook.com/downloads/demos/demo.docx"
+    system "docx2txt.sh", "demo.docx"
```
Use `bin/"docx2txt.sh"` to enure it's using the one we just installed.

dunn commented:
> Error: Testing requires the latest version of docx2txt

Usually that means it's not installed locally.  Did you uninstall it before trying to run the test?

DomT4 commented on Formula/docx2txt.rb:
```diff
@@ -0,0 +1,19 @@
+  url "http://downloads.sourceforge.net/docx2txt/docx2txt-1.4.tgz"
+  sha256 "b297752910a404c1435e703d5aedb4571222bd759fa316c86ad8c8bbe58c6d1b"
+
+  depends_on "perl"
```
Please don't hardcode our `perl`. If it needs a version of Perl a release of OS X doesn't ship use:

```ruby
depends_on :perl => "5.14" if MacOS.version <= :mavericks
```
For example, bending that style of syntax to your needs. If it's fine with OS X's shipped `perl` on OS X as far as you know, you can just remove the dependency completely.

DomT4 commented on Formula/docx2txt.rb:
```diff
@@ -0,0 +1,19 @@
+
+  depends_on "perl"
+  depends_on "unzip"
+  depends_on "bash" => :optional
```
What does this change if it is built with our `bash` rather than the system's?

DomT4 commented on Formula/docx2txt.rb:
```diff
@@ -0,0 +1,19 @@
+  depends_on "bash" => :optional
+
+  def install
+    system "make", "install", "CONFIGDIR=/usr/local/etc"
```
Don't presume `/usr/local`. Should be `CONFIGDIR=#{etc}`

proinsias commented:
> > Error: Testing requires the latest version of docx2txt

> Usually that means it's not installed locally. Did you uninstall it before trying to run the test?

No, though when I try to uninstall, I get `Error: No such keg: /usr/local/Cellar/docx2txt`.

DomT4 commented on Formula/docx2txt.rb:
```diff
@@ -0,0 +1,22 @@
+  end
+
+  def install
+    system "make", "install", "CONFIGDIR=#{etc}"
```
I suspect your errors in running `brew test` here stem from the fact you've not set a prefix here, so it'll be installing into the default prefix rather than the Cellar as intended.

dunn replied:
That would also explain the `Error: No such keg: /usr/local/Cellar/docx2txt` error.

proinsias commented:
Thanks for the tips – this is my first homebrew submission!

Now I'm just left with a testing error. When I run the test I get:

    Check if </tmp/docx2txt-test-20160617-27321-1dgayqb/demo.docx> exists, is readable and has non-zero size!

It seems as if the `demo.docx` is not copied to the `testpath`. I followed the pattern from the `aften.rb` formula. Any ideas?

dunn commented on Formula/docx2txt.rb:
```diff
@@ -0,0 +1,22 @@
+  end
+
+  test do
+    resource("sample_doc").stage testpath
```
This should be `testpath.install resource("sample_doc")`

proinsias commented:
I've changed the method of installing the test resource to `testpath.install resource("sample_doc")`, but still no luck. I've also added a `assert File.exist? "#{testpath}/demo.docx"` statement to verify the file is there. This is now where the test fails – so I'm still not copying the test resource properly.

dunn commented:
Oh, I see the problem.  Homebrew is trying to unzip the `docx`.  You can prevent that though:
```diff
diff --git a/Formula/docx2txt.rb b/Formula/docx2txt.rb
index 8c2d495..9886c33 100644
--- a/Formula/docx2txt.rb
+++ b/Formula/docx2txt.rb
@@ -7,7 +7,7 @@ class Docx2txt < Formula
   depends_on "unzip"

   resource "sample_doc" do
-    url "http://calibre-ebook.com/downloads/demos/demo.docx"
+    url "http://calibre-ebook.com/downloads/demos/demo.docx", :using => :nounzip
     sha256 "269329fc7ae54b3f289b3ac52efde387edc2e566ef9a48d637e841022c7e0eab"
   end
```
Then you can remove the `File.exist?` statements.

proinsias commented:
I've updated. Thanks for the help!

dunn commented:
The test failure now is because (and I should have realized this) `unzip` is not a core formula: http://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/3420/version=el_capitan/console

It's from the homebrew-dupes tap, and we don't allow cross-tap dependencies in core.  Why can't we use the system `unzip`?

proinsias commented:
Removed dependency on `unzip`.

dunn commented:
Thanks! We're so close:
```
        Error: 1 problem in 1 formula
docx2txt:
  * Stable: Please use https:// for http://downloads.sourceforge.net/docx2txt/docx2txt-1.4.tgz

proinsias commented:
Fixed.

dunn commented:
Thanks! Merged as 449314903243247eab5630d0c3918d3a5a914768.
